### DISCUSSION TITLE
Some question about 3D CNN segmentation

### DISCUSSION THREAD

**Discussion by Archaon6734** [Project-MONAI/MONAI#4412] [Q&A]
Hi, I've been study monai for a while , during my study, the following questions arose:
(1)Is 3D CNN segmentation is always better than 2D CNN segmentation?
(2)Monai has implemented a lot of Transforms, but I feel that many of them are performed in 3-dimensional space. It seems that 2D CNN segmentation is not as simple as 3DCNN segmentation. Is it because the design concept of Monai is mainly for 3D CNN segmentation? 
(3)When doing 3D CNN segmentation, will the batch size affect the results? The default normalization method of 3DCNN in Monai is almost always Instance Normalization, which mathematically does not count the mean and variance between batches, 3D CNN segmentation Why take this approach?
（4）If possible, would it be better to use whole volume for segmentation than a patch?
Thanks in advance！

**Comment by ebrahimebrahim** [comment]
I'm no expert, but here are some thoughts:

> (1)Is 3D CNN segmentation is always better than 2D CNN segmentation?

For 3D images, I would say mostly yes, except for the increased model size and gpu memory demands.
2D convolution for 3D images usually means your model gets to see individual slices one at a time, rather than looking at "the whole thing"



> (2)Monai has implemented a lot of Transforms, but I feel that many of them are performed in 3-dimensional space. It seems that 2D CNN segmentation is not as simple as 3DCNN segmentation. Is it because the design concept of Monai is mainly for 3D CNN segmentation?

There are a lot of 3D images in the world of medical imaging.

> （4）If possible, would it be better to use whole volume for segmentation than a patch?

Whole volume is typically better, except for the computing resource cost. How much it matters depends on the problem you are working on. When segmenting lung regions out of chest xrays, I had very little success with a patch based approach. This makes sense, because some global context from the xray is needed for this task. However I imagine that with a skin lesion segmentation task a patch based approach might be just as good as whole-image, since local image information is enough to figure out the task.
